include/boost/url/impl/url.ipp

```diff
@@ -2366,8 +2385,17 @@ ensure_space(
         }
         if( new_cap < n)
             new_cap = n;
+        std::size_t old_cap = cap_;
         s = allocate(new_cap);
         std::memcpy(s, s_, size());
+        auto sp = s_ + old_cap;
+        auto st = detail::segments_table(sp);
+        st.copy(s + cap_, nseg_);
```

Comment:
well you are adding `old_cap` but then using `cap_` why not just leave out the local variable copy and just use `cap_`?

Reply:
Nice. Yes, it's possible to avoid the `old_cap` if we get the references to the tables first.

Before changing this logic, I stored the `old_cap` before because `s = allocate(new_cap);` would invalidate the old `cap_`.